video_xblock/static/js/video_xblock.js

```diff
@@ -46,12 +47,23 @@ function VideoXBlockStudentViewInit(runtime, element) {
       // Discard a message received from another domain
       return;
     try {
+      if (event.data.action === "saveState") {
+        updateTranscriptDownloadUrl(event.data.downloadTranscriptUrl);
+      };
+
       var url = handlers[event.data.action][event.data.xblockUsageId];
       if (url) {
         sendData(url, event.data.info);
       }
     } catch (err){
       console.log(err)
     }
+  };
+  /** Updates transcript download url if it is enabled */
+  function updateTranscriptDownloadUrl(downloadTranscriptUrl) {
+    try {
+      var downloadLinkEl = document.getElementById('download-transcript-link');
+      downloadLinkEl.href = downloadTranscriptHandlerUrl + '?' + downloadTranscriptUrl;
+    } catch (err){}
```

Comment:
w/o try-catch.
```js
var downloadLinkEl = document.getElementById('download-transcript-link');
if (downloadLinkEl) {
  downloadLinkEl.href = downloadTranscriptHandlerUrl + '?' + downloadTranscriptUrl;
}
```
But I believe, it can be overriten to avoid having this if statement. 
You don't need to initialize the plugin if transcripts are disabled. 
You don't even need to add transcript.js in this case.